pkg/controllers/managementapi/usercontrollers/usercontroller_test.go

```diff
@@ -0,0 +1,139 @@
+package usercontrollers
+
+import (
+	"context"
+	"testing"
+
+	apimgmtv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/fakes"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/version"
+)
+
+func newMockUserControllersController(manager *simpleControllerStarter) *userControllersController {
+	return &userControllersController{
+		manager:       manager,
+		clusterLister: &fakes.ClusterListerMock{},
+		clusters: &fakes.ClusterInterfaceMock{
+			ControllerFunc: func() v3.ClusterController {
+				return &fakes.ClusterControllerMock{
+					EnqueueFunc: func(namespace string, name string) {},
+				}
+			},
+			UpdateFunc: func(c *apimgmtv3.Cluster) (*apimgmtv3.Cluster, error) {
+				return c, nil
+			},
+		},
+	}
+}
+
+type simpleControllerStarter struct {
+	startCalled bool
+	stopCalled  bool
+}
+
+func (s *simpleControllerStarter) Start(ctx context.Context, cluster *v3.Cluster, clusterOwner bool) error {
+	s.startCalled = true
+	return nil
+}
+
+func (s *simpleControllerStarter) Stop(cluster *v3.Cluster) {
+	s.stopCalled = true
+}
+
+func TestClusterWithoutControllersVersionAnnotationGetsUpdated(t *testing.T) {
+	t.Parallel()
+	starter := simpleControllerStarter{}
+	controller := newMockUserControllersController(&starter)
+	cluster := &v3.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "my-cluster",
+			Annotations: map[string]string{},
+		},
+		Status: apimgmtv3.ClusterStatus{
+			Version: &version.Info{GitVersion: "1.23.4"},
+		},
+	}
+
+	obj, err := controller.sync("", cluster)
+	require.NoError(t, err)
+	require.NotNil(t, obj)
+	assert.NotEmpty(t, obj.(*v3.Cluster).Annotations[currentClusterControllersVersion])
```

Comment:
Nit: You may want to assert that it's set to a specific value, especially since you do so for other tests.